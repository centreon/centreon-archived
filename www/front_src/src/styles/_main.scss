@import "partials/typography";
@import "partials/fonts";
@import "partials/variables";
@import "partials/messages";
@import "partials/icons";
@import "partials/form/form";
@import "partials/helpers";
@import "partials/mixins";
