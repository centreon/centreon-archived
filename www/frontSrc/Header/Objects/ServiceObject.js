import React from 'react'
import Button from '@material-ui/core/Button'
import Grid from '@material-ui/core/Grid'
import { withStyles } from '@material-ui/core/styles'
import numeral from 'numeral'
import Popover from '@material-ui/core/Popover'
import Typography from '@material-ui/core/Typography'
import ServiceIcon from "../icons/ServiceIcon"

const styles = theme => ({
  root: {
    fontFamily: theme.font.openSans,
    margin: '0 12px',
    '&:before': {
      width: 1,
      height: 30,
      backgroundColor: '#d1d2d4',
      content: '""',
      position: 'absolute',
      top: 18,
      margin: '0px -10px',
    },
  },
  status: {
    margin: '4px',
    color: '#fff',
    width: 38,
    height: 38,
    '& span': {
      fontSize: 16
    },
  },
  errorStatus: {
    margin: '10px 4px',
    width: 46,
    height: 46,
    backgroundColor: theme.palette.error.main,
    '&:hover': {
      backgroundColor: theme.palette.error.light,
    },
    '& span': {
      fontSize: 16,
      color: '#fff',
      fontWeight: '600'
    },

  },
  warningStatus: {
    margin: '10px 4px',
    color: '#fff',
    width: 40,
    height: 40,
    '& span': {
      fontSize: 16
    },
    backgroundColor: theme.palette.warning.main,
    '&:hover': {
      backgroundColor: theme.palette.warning.light,
    },
  },
  unknownStatus: {
    margin: '10px 4px',
    color: '#fff',
    width: 40,
    height: 40,
    '& span': {
      fontSize: 15,
      fontWeight: '600'
    },
    backgroundColor: theme.palette.unknown.main,
    '&:hover': {
      backgroundColor: theme.palette.unknown.dark,
    },
  },
  chip: {
    height: '8px',
    width: '8px',
    borderRadius: 20,
    display: 'inline-table',
    marginRight: 6,
  },
  icon: {
    width: 34,
    height: 34,
    display: 'inline-flex',
    verticalAlign: 'middle',
    margin: '6px',
    color: '#A7A9AC',
    cursor: 'pointer',
    '&:hover': {
      color: '#D1D2D4',
    }
  },
  paper: {
    padding: theme.spacing.unit,
  },
  popover: {
    pointerEvents: 'none',
  },
  objectDetails: {
    padding: '10px 16px',
    borderBottom: '1px solid #d1d2d4',
    '&:last-child' : {
      borderBottom: 'none',
    }
  },
  total: {
    float: 'right',
    marginLeft: 34,
  },
})

const ServiceObject = ({
    classes, object, anchorEl, open, handleClose, handleOpen,
    critical, warning, unknown, ok, pending, total, url
}) => (
  <div className={classes.root}>
      <ServiceIcon
        viewBox="284 -284 600 600"
        className={classes.icon}
        aria-haspopup="true"
        onClick={handleOpen}
      />
      <Button variant="fab" href={critical.url}
              className={(classes.status, classes.errorStatus)}>
        {numeral(critical.unhandled).format('0a')}
      </Button>
      <Button variant="fab" mini  href={warning.url}
              className={( classes.status, classes.warningStatus)}>
        {numeral(warning.unhandled).format('0a')}
      </Button>
      <Button variant="fab" mini color="primary"  href={unknown.url}
              className={(classes.status, classes.unknownStatus)}>
        {numeral(unknown.unhandled).format('0a')}
      </Button>

      <Popover
        open={open}
        anchorEl={anchorEl}
        anchorReference='anchorEl'
        anchorPosition={{ top: 500, left: 400 }}
        onClose={handleClose}
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'left',
        }}
        transformOrigin={{
          vertical: 'top',
          horizontal: 'left',
        }}
      >
        <div className={classes.objectDetails}>
          <Typography variant="caption" gutterBottom>
            <a href={url} title="all services list">
              All services
            </a>
            <span className={classes.total}>{total}</span>
          </Typography>
        </div>
        <div className={classes.objectDetails}>
          <Typography variant="caption" gutterBottom>
            <span className={classes.chip} style={{backgroundColor: '#e00b3d'}}></span>
            <a href={critical.url} title="services critical list">
              {critical.unhandled} Unhandled problems
            </a>
            <span className={classes.total}> / {critical.total}</span>
          </Typography>
        </div>
        <div className={classes.objectDetails}>
          <Typography variant="caption" gutterBottom>
            <span className={classes.chip} style={{backgroundColor: '#FF9A13'}}></span>
            <a href={warning.url} title="services warning list">
              {warning.unhandled} Warning services
            </a>
            <span className={classes.total}> / {warning.total}</span>
          </Typography>
        </div>
        <div className={classes.objectDetails}>
          <Typography variant="caption" gutterBottom>
            <span className={classes.chip} style={{backgroundColor: '#bcbdc0'}}></span>
            <a href={unknown.url} title="services unknown list">
              {unknown.unhandled} Unknown services
            </a>
            <span className={classes.total}> / {unknown.total}</span>
          </Typography>
        </div>
        <div className={classes.objectDetails}>
          <Typography variant="caption" gutterBottom>
            <span className={classes.chip} style={{backgroundColor: '#88b917'}}></span>
            <a href={ok.url} title="services ok list">
              {ok.total} Ok services
            </a>
          </Typography>
        </div>
        {pending.total > 0 ?
          <div className={classes.objectDetails}>
            <Typography variant="caption" gutterBottom>
              <span className={classes.chip} style={{backgroundColor: '#2AD1D4'}}></span>
              <a htef={pending.url} title="pending services list">
                {ok.total} Pending services
              </a>
            </Typography>
          </div> : ''
        }
      </Popover>
  </div>
)

export default withStyles(styles)(ServiceObject)
