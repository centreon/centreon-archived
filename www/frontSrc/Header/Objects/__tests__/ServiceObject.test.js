/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import { shallow, mount, render } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import ServiceObject from '../ServiceObjectContainer'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'

Enzyme.configure({ adapter: new Adapter() })

const theme = createMuiTheme({
  palette: {
    primary: {main: '#88b917', dark: '#597F00'},
    secondary: {main: '#00bfb3'},
    error: {main: '#e00b3d', light: '#ff2c5b', lighter: '#FF8C88'},
    warning: {main: '#ff9a13', light: '#ffb749'},
    unreachable: {main: '#818285', light: '#a6a7a6'},
    unknown: {main: '#bcbdc0', dark: '#95969a'},
    pending: {main: '#2AD1D4'},
  },
  font: {
    openSans: "'Open Sans', Arial, Tahoma, Helvetica, Sans-Serif"
  },
})

describe('ServiceObject', () => {
  const middlewares = [ thunk ]
  const mockStore = configureStore(middlewares)

  const store = mockStore({
    service: {
      url: './main.php?p=20201&o=svc&search=',
      critical: {
        classe: 'error',
        total: '5',
        unhandled: '0',
        url: './main.php?p=20201&o=svc_critical&search='
       },
      warning: {
        classe: 'warning',
        total: '5',
        unhandled: '0',
        url: './main.php?p=20201&o=svc_warning&search='
      },
      ok: {
        classe: 'success',
        total: '58',
        url: './main.php?p=20201&o=svc_ok&search='
      },
      unknown: {
        classe: 'unknown',
        total: '0',
        unhandled: '0',
        url: './main.php?p=20201&o=svc_unknown&search='
      },
      pending: {
        classe: 'pending',
        total: '2',
        url: './main.php?p=20201&o=svc_pending&search='
      },
      error: null,
      isFetching: true,
      refreshTime: 90000,
      total: 150,
      dataFetched: true
    }
  })

  let getServices = jest.fn()

  it('service status renders correctly', () => {
    const component = shallow(<ServiceObject store={store} />)
    expect(toJson(component)).toMatchSnapshot()
  })


  it('should display detail popover when clicking on service icon', () => {
    const component = mount(
      <MuiThemeProvider theme={theme}>
        <ServiceObject store={store} getServices={getServices} />
      </MuiThemeProvider>
    )

    expect(component.find('div#servicePopover').length).toEqual(0)
    component.find('svg#serviceIcon').simulate('click')
    expect(component.find('div#servicePopover').length).toEqual(1)
  })

})