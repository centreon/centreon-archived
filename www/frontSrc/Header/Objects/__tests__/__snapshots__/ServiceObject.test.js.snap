// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceObject service status renders correctly 1`] = `
<ServiceObjectContainer
  getServices={[Function]}
  service={
    Object {
      "critical": Object {
        "classe": "error",
        "total": "5",
        "unhandled": "0",
        "url": "./main.php?p=20201&o=svc_critical&search=",
      },
      "dataFetched": true,
      "error": null,
      "isFetching": true,
      "ok": Object {
        "classe": "success",
        "total": "58",
        "url": "./main.php?p=20201&o=svc_ok&search=",
      },
      "pending": Object {
        "classe": "pending",
        "total": "2",
        "url": "./main.php?p=20201&o=svc_pending&search=",
      },
      "refreshTime": 90000,
      "total": 150,
      "unknown": Object {
        "classe": "unknown",
        "total": "0",
        "unhandled": "0",
        "url": "./main.php?p=20201&o=svc_unknown&search=",
      },
      "url": "./main.php?p=20201&o=svc&search=",
      "warning": Object {
        "classe": "warning",
        "total": "5",
        "unhandled": "0",
        "url": "./main.php?p=20201&o=svc_warning&search=",
      },
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
